### PR TITLE
Harden scripts/run-ui-tests against race conditions

### PR DESCRIPTION
**Pull request by PawelLipski**

### CODE REVIEW

File: scripts/run-ui-tests [hunk 1]
```diff
@@ -5,6 +5,15 @@ set -e -o pipefail -u
 self_dir=$(cd "$(dirname "$0")" &>/dev/null; pwd -P)
 source "$self_dir"/utils.sh
```
Comment (PawelLipski) [inline scripts/run-ui-tests:7]: 3 (locally: `gradle test`, `gradle runIdeForUiTests` -> IDE) or 4 (in CI: `gradle test`, `xvfb-run` -> `gradle runIdeForUiTests` -> IDE) processes in parallel to manage

Comment (PawelLipski) [reply scripts/run-ui-tests:7]: image

Comment (PawelLipski) [reply scripts/run-ui-tests:7]: That's what I liked more about ide-probe (our in-house IDE testing framework): the test suite itself spawned the IDE process

File: scripts/run-ui-tests [hunk 2]
```diff
@@ -22,26 +31,44 @@ function ide_logs_from_this_run() {
 # but unfortunately xvfb-run available for Debian Buster (base of our CI image) doesn't support the latter.
 command=${CI:+xvfb-run --auto-servernum}
 gradle_run_ide_log=~/gradlew-runIdeForUiTests.log
-$command ./gradlew -PintellijVersionForRunIde=$intellij_version --max-workers=1 -Dorg.gradle.jvmargs=-Xmx256M runIdeForUiTests >$gradle_run_ide_log 2>&1 &
+$command ./gradlew -PintellijVersionForRunIde=$intellij_version --no-daemon --max-workers=1 -Dorg.gradle.jvmargs=-Xmx256M runIdeForUiTests >$gradle_run_ide_log 2>&1 &
```
Comment (PawelLipski) [inline scripts/run-ui-tests:34]: At least one process to care about less: Gradle daemon

File: uiTests/src/test/resources/ide.rhino.js [hunk 3]
```diff
@@ -41,16 +41,6 @@ function Ide() {
       });
     });
   };
-
-  this.close = function () {
-    // We can't just run `app.exit()` synchronously since then the Robot client would never receive a reply from the Robot server,
```
Comment (PawelLipski) [inline uiTests/src/test/resources/ide.rhino.js:46]: This was a dumb idea from the very beginning, `Do Not Try This At Home` :hushed:

Comment (PawelLipski) [reply uiTests/src/test/resources/ide.rhino.js:46]: In fact now I observe that simply sending TERM to the wrapping Gradle/xvfb-run process makes the IDE terminate more clearly than with this sh*t :hankey:

File: .circleci/config.yml [hunk 4]
```diff
@@ -55,6 +55,9 @@ jobs:
           name: Verify binary compatibility with supported IntelliJ versions
           command: ./scripts/verify-binary-compatibility
 
+      - run:
+          name: '[TEMP] Install prerequisites of UI tests'
```
Comment (micpiotrowski) [inline .circleci/config.yml:59]: Docker image regenerated

**Pull request merged**